### DISCUSSION TITLE
Problem with Zigbee and CC2530

### DISCUSSION THREAD

**Discussion by Hans0612** [arendst/Tasmota#15298] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Zigbee won't start      
received bad FCS frame
 and reported {"Status":99,"Message":"Abort"}

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ x] Device used (e.g., Sonoff Basic): ___Wemos D1 Mini pro /Wemos D1 Mini__
- [ x] Tasmota binary firmware version number used: ____11.0.0.5_
  - [x ] Pre-compiled
  - [ ] Self-compiled
- [ X] Flashing tools used: _Tasmotizer / SmartRF Flash Programmer_
- [ X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:

10:59:46.554 CMD: Backlog Template; Module; GPIO 255
10:59:46.556 SRC: WebConsole from 172.172.X.X
10:59:46.558 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 26, Pld -99, Data 'Template; Module; GPIO 255'
10:59:46.636 SRC: Backlog
10:59:46.638 CMD: Grp 0, Cmd 'TEMPLATE', Idx 1, Len 0, Pld -99, Data ''
10:59:46.641 RSL: RESULT = {"NAME":"Zigbee","GPIO":[0,0,0,0,0,0,0,0,0,3584,0,3552,0,0],"FLAG":0,"BASE":18}
10:59:46.845 SRC: Backlog
10:59:46.847 CMD: Grp 0, Cmd 'MODULE', Idx 1, Len 0, Pld -99, Data ''
10:59:46.849 RSL: RESULT = {"Module":{"0":"Zigbee"}}
10:59:47.097 SRC: Backlog
10:59:47.099 CMD: Grp 0, Cmd 'GPIO', Idx 1, Len 3, Pld 255, Data '255'
10:59:47.104 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"3584":"Zigbee Rx"},"GPIO14":{"0":"None"},"GPIO15":{"3552":"Zigbee Tx"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
11:00:01.497 WIF: Checking connection...
11:00:21.463 WIF: Checking connection...

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ X] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
11:02:20.454 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 0, Data '0'
11:02:20.457 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_55DE26","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
11:02:20.461 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:11:43","StartupUTC":"2022-04-05T09:50:37","Sleep":50,"CfgHolder":4617,"BootCount":10,"BCResetTime":"2022-04-05T10:34:57","SaveCount":19,"SaveAddress":"3FB000"}}
11:02:20.465 RSL: STATUS2 = {"StatusFWR":{"Version":"11.0.0.5(tasmota)","BuildDateTime":"2022-04-04T18:12:05","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":160,"Hardware":"ESP8266EX","CR":"331/699"}}
11:02:20.468 RSL: STATUS3 = {"StatusLOG":{"SerialLog":4,"WebLog":4,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Visam-SI",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006000","00004000"]}}
11:02:20.475 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":758,"Free":1288,"Heap":24,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04768001","000000CF","010013C0","C000F981","00004084","00201000","04000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,23,24,26,27,29,30,35,37,38,45,46,50,56,62","Sensors":"1,2,3,4,5,6"}}
11:02:20.479 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-55DE26-7718","IPAddress":"172.172.X.X","Gateway":"172.172.X.X","Subnetmask":"255.255.0.0","DNSServer1":"172.172.X.X","DNSServer2":"0.0.0.0","Mac":"C4:5B:BE:55:DE:26","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
11:02:20.482 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_55DE26","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
11:02:20.486 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-04-05T10:02:20","Local":"2022-04-05T11:02:20","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:19","Sunset":"19:26"}}
11:02:20.489 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-04-05T11:02:20"}}
11:02:20.492 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-04-05T11:02:20","Uptime":"0T00:11:43","UptimeSec":703,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"XXXXXXXX","BSSId":"00:27:22:26:AB:0B","Channel":12,"Mode":"11n","RSSI":36,"Signal":-82,"LinkCount":1,"Downtime":"0T00:00:03"}}}
11:02:21.471 WIF: Checking connection...

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

00:00:00.002 HDW: ESP8266EX
00:00:00.011 UFS: FlashFS mounted with 1992 kB free
00:00:00.050 CFG: Loaded from File, Count 19
00:00:00.053 QPC: Count 1
00:00:00.055 CFG: CR 331/699, Busy 0
00:00:00.058 ROT: Mode 1
00:00:00.059 ZIG: GPIOs Rx:13 Tx:15
00:00:00.060 SRC: Restart
00:00:00.062 Project tasmota - Tasmota Version 11.0.0.5(tasmota)-2_7_4_9(2022-04-04T18:12:05)
00:00:00.215 WIF: Checking connection...
00:00:00.216 WIF: Attempting connection...
00:00:00.552 WIF: Connecting to AP1 XXXXXXX Channel 12 BSSId 00:27:22:26:AB:0B in mode 11n as tasmota-55DE26-7718...
00:00:02.472 WIF: Checking connection...
00:00:02.473 WIF: Connected
00:00:02.674 HTP: Web server active on tasmota-55DE26-7718 with IP address 172.172.X.X
00:00:02.927 HTP: Main Menu
00:00:03.415 WIF: Sending Gratuitous ARP
00:00:03.416 NTP: Sync time...
00:00:03.498 RTC: UTC 2022-04-05T10:10:44, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
11:10:44.000 RTC: Synced by NTP
11:10:44.205 RSL: INFO1 = {"Info1":{"Module":"Zigbee","Version":"11.0.0.5(tasmota)","FallbackTopic":"cmnd/DVES_55DE26_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:10:44.209 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-55DE26-7718","IPAddress":"172.172.X.X"}}
11:10:44.219 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":11}}
11:10:44.231 TFS: File 'autoexec.bat' not found
11:10:46.557 HTP: Consoles
11:10:47.431 QPC: Reset
11:10:48.259 HTP: Console
11:10:49.415 APP: Boot Count 11
11:10:49.418 RSL: STATE = {"Time":"2022-04-05T11:10:49","Uptime":"0T00:00:10","UptimeSec":10,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"Visam-SI","BSSId":"00:27:22:26:AB:0B","Channel":12,"Mode":"11n","RSSI":38,"Signal":-81,"LinkCount":1,"Downtime":"0T00:00:04"}}
11:10:50.370 CFG: Saved to flash at 3FB, Count 20, Bytes 4096
11:10:56.057 ZIG: rebooting ZNP device
11:10:56.229 ZIG: ZbZNPSent 0xFF x 255
11:10:56.230 ZIG: ZbZNPSent 410000
11:10:57.858 ZbZNPReceived: received bad FCS frame FE1F48805465786173496E7472756D656E7320202053616D706C65204C69676874B2FE17, 238
11:10:57.907 ZbInput discarding byte 48
11:10:57.908 ZbInput discarding byte 80
11:10:57.908 ZbInput discarding byte 49
11:10:57.909 ZbInput discarding byte 45
11:10:57.911 ZbInput discarding byte 45
11:10:57.915 ZbInput discarding byte 45
11:10:57.918 ZbInput discarding byte 3A
11:10:57.922 ZbInput discarding byte 20
11:10:57.925 ZbInput discarding byte 30
11:10:57.929 ZbInput discarding byte 30
11:10:57.934 ZbInput discarding byte 31
11:10:57.936 ZbInput discarding byte 32
11:10:57.939 ZbInput discarding byte 34
11:10:57.943 ZbInput discarding byte 42
11:10:57.947 ZbInput discarding byte 30
11:10:57.950 ZbInput discarding byte 30
11:10:57.954 ZbInput discarding byte 31
11:10:57.957 ZbInput discarding byte 46
11:10:57.961 ZbInput discarding byte 43
11:10:57.964 ZbInput discarding byte 36
11:10:57.968 ZbInput discarding byte 34
11:10:57.972 ZbInput discarding byte 35
11:10:57.975 ZbInput discarding byte 32
11:10:57.979 ZbInput discarding byte 45
11:10:57.982 ZbInput discarding byte E8
11:10:57.986 ZIG: {"ZbZNPReceived":"4880494545453A202030303132344230303146433634353245"}
11:10:57.996 ZIG: {"ZbZNPReceived":"4880526F757465723A204435423120506172656E743A2030"}
11:10:58.044 ZbZNPReceived: received bad FCS frame FE1D4880526F757465723A20443531205357353A2048656C7020202020202020FCFE, 188
11:10:58.093 ZbInput discarding byte 1D
11:10:58.094 ZbInput discarding byte 48
11:10:58.095 ZbInput discarding byte 80
11:10:58.096 ZbInput discarding byte 52
11:10:58.097 ZbInput discarding byte 6F
11:10:58.101 ZbInput discarding byte 75
11:10:58.105 ZbInput discarding byte 74
11:10:58.108 ZbInput discarding byte 65
11:10:58.112 ZbInput discarding byte 72
11:10:58.115 ZbInput discarding byte 3A
11:10:58.119 ZbInput discarding byte 20
11:10:58.123 ZbInput discarding byte 44
11:10:58.126 ZbInput discarding byte 35
11:10:58.129 ZbInput discarding byte 42
11:10:58.134 ZbInput discarding byte 31
11:10:58.137 ZbInput discarding byte 20
11:10:58.140 ZbInput discarding byte 53
11:10:58.144 ZbInput discarding byte 57
11:10:58.148 ZbInput discarding byte 35
11:10:58.151 ZbInput discarding byte 3A
11:10:58.154 ZbInput discarding byte 20
11:10:58.158 ZbInput discarding byte 48
11:10:58.162 ZbInput discarding byte 65
11:10:58.165 ZbInput discarding byte 6C
11:10:58.169 ZbInput discarding byte 70
11:10:58.172 ZbInput discarding byte 20
11:10:58.176 ZbInput discarding byte 20
11:10:58.179 ZbInput discarding byte 20
11:10:58.183 ZbInput discarding byte 20
11:10:58.187 ZbInput discarding byte 20
11:10:58.190 ZbInput discarding byte 20
11:10:58.194 ZbInput discarding byte 20
11:10:58.197 ZbInput discarding byte FC
11:11:01.256 ZIG: timeout, goto label 99
11:11:01.259 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
11:11:01.261 ZIG: Abort
11:11:01.261 ZIG: Stopping (99)
11:11:03.459 WIF: Checking connection...

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
You may not have done a correct flashing of the needed CC2530 firmware. It won't work "out of the box".

**Reply by sfromis** [reply]
For flashing a similar CC2531, I used the "Firmware Upgrade" function in Tasmota, after configuring a template like this:
```
{"NAME":"CCLoader","GPIO":[32,0,320,0,0,0,0,0,3584,3552,5312,5731,0,0],"FLAG":0,"BASE":18}
Gpio.00    32 Button1
Gpio.01     0 None
Gpio.02   320 Led_i1
Gpio.03     0 None
Gpio.04     0 None
Gpio.05     0 None
Gpio.09     0 None
Gpio.10     0 None
Gpio.12  3584 Zigbee Rx
Gpio.13  3552 Zigbee Tx
Gpio.14  5312 Zigbee Rst
Gpio.15  5731 Option A4
Gpio.16     0 None
Gpio.17     0 None
```
And obviously wiring to match.

"Option A4" is to "Enable CCLoader using Zigbee Rx/Tx/Rst Gpios". Button/led config on gpio0/2 are not for flashing. Pin selection was to be able to have a 8-pin connector directly fitting a NodeMCU. Gpio15 needs no connection to anything.

**Reply by Hans0612** [reply]
@sfromis 
I used the CC Debugger for firmware flash (CC2530ZNP-Prod.bin).
Shouldn't that work too?!
Flashed two CC2530 and also several times, but the error still remained the same....
I haven't managed to flash with Tasmota yet, I took Wemos D1 Mini for it, but it should also work. I would have to use the Tasmota zigbee.bin to flash and only adjust the template, right?

**Reply by sfromis** [reply]
Not being well-versed in the details, I can't tell if some combination "should work" or not. You told nothing about where you got that `CC2530ZNP-Prod.bin`, this may make a difference.

The Tasmota flashing scenario I used was by grabbing the `.hex` from [Z-Stack](https://github.com/Koenkk/Z-Stack-firmware/), and doing an objcopy to generate a `.bin` file for flashing, which IIRC is *not the same* as you'll find on Z-Stack.
`objcopy --gap-fill 0xFF --pad-to 0x040000 -I ihex inputfile.hex -O binary outputfile.bin`

Personally, I considered myself lucky when I saw it working :-D I don't know what there may (not not) be of caveats for using CC Debugger, I just did what to me looked like the straightforward path, in a Tasmota context.

Here, you have some `.bin` files already made ready for ccloader flashing:
https://github.com/arendst/Tasmota/tree/development/tools/fw_Zigbee_cc2530
